langchain4j-azure-ai-search/src/main/java/dev/langchain4j/store/embedding/azure/search/AzureAiSearchEmbeddingStore.java

```diff
@@ -0,0 +1,426 @@
+        String id = randomUUID();
+        addInternal(id, embedding, textSegment);
+        return id;
+    }
+
+    /**
+     * Adds multiple embeddings to the store.
+     *
+     * @param embeddings A list of embeddings to be added to the store.
+     * @return A list of auto-generated IDs associated with the added embeddings.
+     */
+    @Override
+    public List<String> addAll(List<Embedding> embeddings) {
+        List<String> ids = embeddings.stream().map(ignored -> randomUUID()).collect(toList());
+        addAllInternal(ids, embeddings, null);
+        return ids;
+    }
+
+    /**
+     * Adds multiple embeddings and their corresponding contents that have been embedded to the store.
+     *
+     * @param embeddings A list of embeddings to be added to the store.
+     * @param embedded   A list of original contents that were embedded.
+     * @return A list of auto-generated IDs associated with the added embeddings.
+     */
+    @Override
+    public List<String> addAll(List<Embedding> embeddings, List<TextSegment> embedded) {
+        List<String> ids = embeddings.stream().map(ignored -> randomUUID()).collect(toList());
+        addAllInternal(ids, embeddings, embedded);
+        return ids;
+    }
+
+    /**
+     * Finds the most relevant (closest in space) embeddings to the provided reference embedding.
+     *
+     * @param referenceEmbedding The embedding used as a reference. Returned embeddings should be relevant (closest) to this one.
+     * @param maxResults         The maximum number of embeddings to be returned.
+     * @param minScore           The minimum relevance score, ranging from 0 to 1 (inclusive).
+     *                           Only embeddings with a score of this value or higher will be returned.
+     * @return A list of embedding matches.
+     * Each embedding match includes a relevance score (derivative of cosine distance),
+     * ranging from 0 (not relevant) to 1 (highly relevant).
+     */
+    @Override
+    public List<EmbeddingMatch<TextSegment>> findRelevant(Embedding referenceEmbedding, int maxResults, double minScore) {
+        List<Float> vector = floatsArrayToList(referenceEmbedding.vector());
+        VectorizedQuery vectorizedQuery = new VectorizedQuery(vector)
+                .setFields(DEFAULT_FIELD_CONTENT_VECTOR)
+                .setKNearestNeighborsCount(maxResults);
+
+        SearchPagedIterable searchResults =
+                searchClient.search(null,
+                        new SearchOptions()
+                                .setVectorSearchOptions(new VectorSearchOptions().setQueries(vectorizedQuery)),
+                        Context.NONE);
+
+        List<EmbeddingMatch<TextSegment>> result = new ArrayList<>();
+        for (SearchResult searchResult : searchResults) {
+            Double score = fromAzureScoreToRelevanceScore(searchResult.getScore());
+            if (score < minScore) {
+                break;
```

Comment:
should this be `continue`? or Azure always returns highest score first?

Reply:
I would guess so, but I couldn't find it in the documentation, so migrating to "continue" to be safe